address_alias.go

```diff
@@ -0,0 +1,162 @@
+	AliasAddressSerializedBytesSize = serializer.SmallTypeDenotationByteSize + AliasAddressBytesLength
+)
+
+var (
+	emptyAliasAddress = [20]byte{}
+)
+
+// ParseAliasAddressFromHexString parses the given hex string into an AliasAddress.
+func ParseAliasAddressFromHexString(hexAddr string) (*AliasAddress, error) {
+	addrBytes, err := DecodeHex(hexAddr)
+	if err != nil {
+		return nil, err
+	}
+	addr := &AliasAddress{}
+	copy(addr[:], addrBytes)
+	return addr, nil
+}
+
+// MustParseAliasAddressFromHexString parses the given hex string into an AliasAddress.
+// It panics if the hex address is invalid.
+func MustParseAliasAddressFromHexString(hexAddr string) *AliasAddress {
+	addr, err := ParseAliasAddressFromHexString(hexAddr)
+	if err != nil {
+		panic(err)
+	}
+	return addr
+}
+
+// AliasAddress defines an Alias address.
+// An AliasAddress is the Blake2b-160 hash of the OutputID which created it.
+type AliasAddress [AliasAddressBytesLength]byte
+
+func (aliasAddr *AliasAddress) Clone() Address {
+	cpy := &AliasAddress{}
+	copy(cpy[:], aliasAddr[:])
+	return cpy
+}
+
+func (aliasAddr *AliasAddress) VByteCost(costStruct *RentStructure, _ VByteCostFunc) uint64 {
+	return costStruct.VBFactorData.Multiply(serializer.SmallTypeDenotationByteSize + AliasAddressBytesLength)
+}
+
+func (aliasAddr *AliasAddress) Key() string {
+	return string(append([]byte{byte(AddressAlias)}, (*aliasAddr)[:]...))
+}
+
+func (aliasAddr *AliasAddress) Chain() ChainID {
+	return AliasID(*aliasAddr)
+}
+
+func (aliasAddr *AliasAddress) AliasID() AliasID {
+	return AliasID(*aliasAddr)
+}
+
+func (aliasAddr *AliasAddress) Equal(other Address) bool {
+	otherAddr, is := other.(*AliasAddress)
+	if !is {
+		return false
+	}
+	return *aliasAddr == *otherAddr
+}
+
+func (aliasAddr *AliasAddress) Type() AddressType {
+	return AddressAlias
+}
+
+func (aliasAddr *AliasAddress) Bech32(hrp NetworkPrefix) string {
+	return bech32String(hrp, aliasAddr)
+}
+
+func (aliasAddr *AliasAddress) String() string {
+	return EncodeHex(aliasAddr[:])
+}
+
+func (aliasAddr *AliasAddress) Deserialize(data []byte, deSeriMode serializer.DeSerializationMode, deSeriCtx interface{}) (int, error) {
+	if deSeriMode.HasMode(serializer.DeSeriModePerformValidation) {
+		if err := serializer.CheckMinByteLength(AliasAddressSerializedBytesSize, len(data)); err != nil {
+			return 0, fmt.Errorf("invalid alias address bytes: %w", err)
+		}
+		if err := serializer.CheckTypeByte(data, byte(AddressAlias)); err != nil {
+			return 0, fmt.Errorf("unable to deserialize alias address: %w", err)
+		}
+	}
+	copy(aliasAddr[:], data[serializer.SmallTypeDenotationByteSize:])
+	return AliasAddressSerializedBytesSize, nil
+}
+
+func (aliasAddr *AliasAddress) Size() int {
+	return AliasAddressSerializedBytesSize
+}
+
+func (aliasAddr *AliasAddress) Serialize(_ serializer.DeSerializationMode, deSeriCtx interface{}) (data []byte, err error) {
```

Comment:
Move that function over `Size()`, then its the same order like in `address_nft.go` and `address_ed25519.go`